### PR TITLE
use guzzle globally through a helper

### PR DESCRIPTION
**Pull request by individual-it**
## Description
1. use Guzzle through a Helper class
2. don't catch Guzzle exceptions all over the place, as there are not thrown any-more, but only check responses

## Related Issue
https://github.com/owncloud/QA/issues/586

## Motivation and Context
have only one place where guzzle can be configured globally

## How Has This Been Tested?
:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/TestHelpers/HttpRequestHelper.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+namespace TestHelpers;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+
+class HttpRequestHelper {
+	public static function sendRequest(
+		$url, $method='GET', $user=null, $password=null, $headers=null, $body = null, $config = null, $cookies = null
+	) {
+		$client = new Client();
```
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/TestHelpers/HttpRequestHelper.php]: Maybe, but what for?
The only thing that I can see that would make sense is the base part of the URL

[the remark this answers is not in the document]
Comment (patrickjahns) [reply tests/TestHelpers/HttpRequestHelper.php]: Yes exactly - how do we define the server url currently?

**Comment by patrickjahns** [comment]
Wondering if we should add a unit test to ensure the clientHelper works as designed?

File: tests/TestHelpers/HttpRequestHelper.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (individual-it) [reply tests/TestHelpers/HttpRequestHelper.php]: from wherever `sendRequest()` is called is responsible to create the complete URL

File: tests/TestHelpers/TagsHelper.php [hunk 2]
```diff
@@ -178,10 +177,6 @@ public static function createTag(
 			$davPathVersionToUse,
 			"systemtags"
 		);
-		$responseHeaders = $response->getHeaders();
-		$tagUrl = $responseHeaders['Content-Location'][0];
-		$lastTagId = \substr($tagUrl, \strrpos($tagUrl, '/') + 1);
-		return ['lastTagId' => $lastTagId, 'HTTPResponse' => $response];
```
Comment (individual-it) [inline tests/TestHelpers/TagsHelper.php]: moved to `Tags.php`

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 3]
```diff
@@ -337,20 +310,19 @@ private function checkUserDownload($url, $options, $mimeType) {
...
-	private function checkDownload($url, $auth = null, $mimeType = null) {
-		if ($auth !== null) {
-			$options['auth'] = $auth;
-		}
-		$options['stream'] = true;
-		$options['headers']['X-Requested-With'] = 'XMLHttpRequest';
-
-		$client = new Client();
-		$this->response = $client->get($url, $options);
+	private function checkDownload(
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/Sharing.php]: had to change that function to get the nice username & password

**Comment by individual-it** [comment]
@phil-davis  @patrickjahns please review 
signal-attachment-2018-08-01-201208

File: tests/acceptance/features/bootstrap/Auth.php [hunk 4]
```diff
@@ -177,15 +164,13 @@ public function aNewAppPasswordHasBeenGenerated($name) {
...
-					'user' => $user,
-					'password' => $this->getPasswordForUser($user),
-			]
-			]
-		);
+		$body = \json_encode([
+			'user' => $user,
+			'password' => $this->getPasswordForUser($user),
+		]);
+		$headers = ['Content-Type' => 'application/json'];
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Auth.php]: I guess the previous ``$client->post()`` with that ``json`` array of values had been automagically setting this header.

**Review by phil-davis — APPROVED** [review]
This looks like a great thing. It should help with the Guzzle6 PR, reducing the diff list a lot.

**Comment by phil-davis** [comment]
I pushed a commit with code style changes to make our "informal" ``phpcs`` pass.
And added some missing spaces either side of ``=`` signs.

I couldn't see any issues with the actual code refactoring - good work.

File: tests/acceptance/features/bootstrap/CalDavContext.php [hunk 5]
```diff
@@ -174,8 +174,8 @@ public function userHasCreatedACalendarNamed($user, $name) {
 		$davUrl = $this->featureContext->getBaseUrl()
 			. "/remote.php/dav/calendars/$user/$name";
 
-		$body =
-			'<c:mkcalendar xmlns:c="urn:ietf:params:xml:ns:caldav" ' .
+		$body
+			= '<c:mkcalendar xmlns:c="urn:ietf:params:xml:ns:caldav" ' .
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/CalDavContext.php]: I think the previous code looks fine, but ``phpcs`` has a different idea.

**Comment by individual-it** [comment]
squashed and pushed again.
@patrickjahns for the unit tests I opened a new issue https://github.com/owncloud/core/issues/32527

**Comment by individual-it** [comment]
backport in  #32528

**Pull request merged**